source/scale/high-availability-cluster-based-deployment.rst

```diff
@@ -8,462 +8,414 @@ High availability cluster-based deployment
+For improved performance, apply the following recommendations:
 
-On large deployments, also consider using the :ref:`search replicas <configure/environment-configuration-settings:search replicas>` feature to isolate search queries onto one or more search replicas. A search replica is similar to a read replica, but is used only for handling search queries.
+Primary Node:
 
-.. note::
+- ``max_connections = 1024``
+- ``random_page_cost = 1.1``
+- ``effective_cache_size = 21GB``
+- ``shared_buffers = 21GB``
+- ``tcp_keepalives_idle = 5``
+- ``tcp_keepalives_interval = 1``
+- ``tcp_keepalives_count = 5``
+- ``maintenance_work_mem = 512MB``
+- ``autovacuum_max_workers = 4``
+- ``autovacuum_vacuum_cost_limit = 500``
+- ``max_worker_processes = 12``
+- ``max_parallel_workers_per_gather = 4``
+- ``max_parallel_workers = 12``
+- ``max_parallel_maintenance_workers = 4``
 
-  For an AWS High Availability RDS cluster deployment, don't hard-code the IP addresses. Point this configuration setting directly to the underlying read-only node endpoints within the RDS cluster. We recommend circumventing the failover/load balancing that AWS/RDS takes care of (except for the write traffic). Mattermost has its own method of balancing the read-only connections, and can also balance those queries to the DataSource/write+read connection should those nodes fail. 
+Replica Node (in addition to the primary node settings above):
 
-Mattermost distributes queries as follows:
+- ``work_mem = 16MB``
+- ``hot_standby = on``
+- ``hot_standby_feedback = on``
 
-* All write requests, and some specific read requests, are sent to the master.
-* All other read requests (excluding those specific queries that go to the master) are distributed among the available read replicas. If no read replicas are available, these are sent to the master instead.
-* Search requests are distributed among the available search replicas. If no search replicas are available, these are sent to the read replicas instead (or, if no read replicas are available, to the master).
+File storage
+~~~~~~~~~~~~
 
-Size the databases
-``````````````````
+Proper shared storage setup for files (e.g., NAS/S3) is another fundamental requirement before progressing to configuring the application.
 
-For information about sizing database servers, see :ref:`hardware-sizing-for-enterprise`.
+For shared storage:
 
-In a master/slave environment, make sure to size the slave machine to take 100% of the load in the event that the master machine goes down and you need to fail over.
+- Use NAS or S3 for ``DriverName``: ``local``. Migrating to NAS or S3 from local storage is beyond the scope of this document.
```

Comment:
This is incorrect. If `DriverName` is `local`, then a local filesystem is assumed, not S3.